[FSDP] Clarify CPU offload implicitly in reshard_doc

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98667
* __->__ #98666

Per title

Differential Revision: [D44812344](https://our.internmc.facebook.com/intern/diff/D44812344/)

rohan-varma committed:
[FSDP] Clarify CPU offload implicitly in reshard_doc

Per title

Differential Revision: [D44812344](https://our.internmc.facebook.com/intern/diff/D44812344/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/flat_param.py:
```diff
@@ -1487,7 +1487,9 @@ def reshard(self, free_unsharded_flat_param: bool):
-        sharded flat parameter.
+        sharded flat parameter. Note that this also implicitly offloads
+        the sharded flat parameter (if CPU offload is enabled) by pointing
+        it to the local_shard attribute which resides on CPU.
```
nit: Should we specify _local_shard with the prefixed underscore? (backticks do not matter much can be can do \'\'_local_shard\'\' for consistency)

awgu reviewed (approved):
Thanks for clarifying!

rohan-varma committed:
Update on "[FSDP] Clarify CPU offload implicitly in reshard_doc"


Per title

Differential Revision: [D44812344](https://our.internmc.facebook.com/intern/diff/D44812344/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f "Doc only"
